react-frontend-app/src/components/BarGraph.js

```diff
@@ -0,0 +1,217 @@
+            .style("font-size", "16px") 
+            .style("font-weight", "bold")
+            .text("Full Race Prediction Model Results");
+
+        // Create bars
+        var bar = svg.selectAll("g")
+            .data(zippedData)
+            .enter()
+            .append("g")
+            .attr("transform", function(d, i) { 
+                return "translate(" + spaceForLabels + "," + (i * barHeight + gapBetweenGroups * (0.5 + Math.floor(i/2))) + ")";
+            });
+        
+        // Create rectangles of the correct width
+        bar.append("rect")
+            .attr("fill", function(d,i) { return color(i % 2); })
+            .attr("class", "bar")
+            .attr("width", x)
+            .attr("height", barHeight - 1);
+
+        // Add text label in bar
+        bar.append("text")
+            .attr("x", function(d) { return x(d) + 5; })
+            .attr("y", barHeight / 2)
+            .attr("font-size", "10px")
+            .attr("fill", "black")
+            .attr("dy", ".35em")
+            .text(function(d) { return d3.format(".2%")(d); });
+
+        // Draw labels
+        bar.append("text")
+            .attr("class", "label")
+            .attr("x", function(d) { return - 20; })
+            .attr("text-anchor", "end")  
+            .attr("y", groupHeight / 2)
+            .attr("dy", ".35em")
+            .text(function(d,i) {
+                if (i % 2 === 0)
+                    return racesCaps[Math.floor(i/2)];
+                else
+                    return "";
+            });
+
+        // Display y-axis
+        svg.append("g")
+            .attr("class", "y axis")
+            .attr("transform", "translate(" + spaceForLabels + ", " + -gapBetweenGroups/2 + ")")
+            .call(yAxis);
+        
+        // Draw legend/key
+        var legendRectSize = 18,
+            legendSpacing  = 4;
+
+        var legend = svg.selectAll('.legend')
+            .data(["BISG Model", "Zest Model"])
+            .enter()
+            .append('g')
+            .attr('transform', function (d, i) {
+                var height = legendRectSize + legendSpacing;
+                var offset = -gapBetweenGroups/2;
+                var horz = spaceForLabels + chartWidth + 100 - legendRectSize;
+                var vert = i * height - offset;
+                return 'translate(' + horz + ',' + vert + ')';
+            });
+
+        legend.append('rect')
+            .attr('width', legendRectSize)
+            .attr('height', legendRectSize)
+            .style('fill', function (d, i) { return color(i); })
+            .style('stroke', function (d, i) { return color(i); });
+
+        legend.append('text')
+            .attr('class', 'legend')
+            .attr('x', legendRectSize + legendSpacing + 3)
+            .attr('y', legendRectSize - legendSpacing)
+            .attr("font-size", "12px")
+            .attr("fill", "black")
+            .text(function (d) { return d; });
+
```

Comment:
Outstanding! Your code is super legible, really clean, fantastic design! Great job all around!

Reply:
Second this!!

Reply:
Thirded^